src/layout/verticalPanel/searchResultList.ts

```diff
@@ -410,13 +409,13 @@ export class SearchResultList extends St.BoxLayout {
         }
         provider.getResultMetas(
             results,
-            (resMetas, index) => {
+            (resMetas) => {
                 this.resMetas = resMetas;
-                let moreEntry;
+                let moreEntry: SearchResultEntry | null = null;
                 //
-                const extraResults = [];
+                const extraResults: SearchResultEntry[] = [];
                 if (resMetas.length > 5) {
-                    moreEntry = new SearchResultEntry(
+                    const more = moreEntry = new SearchResultEntry(
```

Comment:
Variable duplication there?

Reply:
To make typescript realize that the variable cannot be null inside the callback.